tests/test_set_evidence.py

```diff
@@ -0,0 +1,818 @@
+
+
+def test_data_exfiltration(set_evidence_helper, db_mock):
+    """Testing the data_exfiltration method."""
+    set_evidence_helper.data_exfiltration(
+        daddr="10.0.0.1",
+        src_mbs=100.0,
+        profileid="profile_192.168.0.1",
+        twid="timewindow11",
+        uid=["unique_id"],
+        timestamp="2023-05-06T12:00:00Z",
+    )
+
+    assert db_mock.set_evidence.call_count == 2
+    args, _ = db_mock.set_evidence.call_args_list[0]
+    evidence = args[0]
+    assert evidence.evidence_type == EvidenceType.DATA_UPLOAD
+    assert evidence.attacker.value == "192.168.0.1"
+    assert evidence.threat_level == ThreatLevel.INFO
+    assert evidence.category == IDEACategory.MALWARE
+    assert evidence.profile.ip == "192.168.0.1"
+    assert evidence.timewindow.number == 11
+    assert evidence.uid == ["unique_id"]
+    assert evidence.source_target_tag == Tag.ORIGIN_MALWARE
+
+
+def test_bad_smtp_login(set_evidence_helper, db_mock):
+    """Testing the bad_smtp_login method."""
+    set_evidence_helper.bad_smtp_login(
+        saddr="192.168.0.1",
+        daddr="10.0.0.1",
+        stime="2023-05-06T12:00:00Z",
+        twid="timewindow12",
+        uid="unique_id",
+    )
+
+    assert db_mock.set_evidence.call_count == 1
+    args, _ = db_mock.set_evidence.call_args
+    evidence = args[0]
+    assert evidence.evidence_type == EvidenceType.BAD_SMTP_LOGIN
+    assert evidence.attacker.value == "192.168.0.1"
+    assert evidence.victim.value == "10.0.0.1"
+    assert evidence.threat_level == ThreatLevel.HIGH
+    assert evidence.category == IDEACategory.ATTEMPT_LOGIN
+    assert evidence.profile.ip == "192.168.0.1"
+    assert evidence.timewindow.number == 12
+    assert evidence.uid == ["unique_id"]
+
+
+def test_smtp_bruteforce(set_evidence_helper, db_mock):
+    """Testing the smtp_bruteforce method."""
+    flow = {
+        "saddr": "192.168.0.1",
+        "daddr": "10.0.0.1",
+        "starttime": "2023-05-06T12:00:00Z",
+    }
+    twid = "timewindow13"
+    uid = ["unique_id1", "unique_id2"]
+    smtp_bruteforce_threshold = 10
+    set_evidence_helper.smtp_bruteforce(
+        flow=flow,
+        twid=twid,
+        uid=uid,
+        smtp_bruteforce_threshold=smtp_bruteforce_threshold,
+    )
+    assert db_mock.set_evidence.call_count == 1
+    args, _ = db_mock.set_evidence.call_args
+    evidence = args[0]
+    assert evidence.evidence_type == EvidenceType.SMTP_LOGIN_BRUTEFORCE
+    assert evidence.attacker.value == "192.168.0.1"
+    assert evidence.victim.value == "10.0.0.1"
+    assert evidence.threat_level == ThreatLevel.HIGH
+    assert evidence.category == IDEACategory.ATTEMPT_LOGIN
+    assert evidence.profile.ip == "192.168.0.1"
+    assert evidence.timewindow.number == 13
+    assert sorted(evidence.uid) == sorted(uid)
+    assert evidence.conn_count == smtp_bruteforce_threshold
+
+
+def test_malicious_ssl(set_evidence_helper, db_mock):
+    """Testing the malicious_ssl method."""
```

Comment:
2 evidence here and youre only checking for 1

Reply:
okay I will make this changes and then update this PR